samples/boards/npcx7m6fb_evb/demo/src/main.c

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright (c) 2020 Nuvoton Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <device.h>
+#include <devicetree.h>
+#include <drivers/gpio.h>
+#include <drivers/uart.h>
+#include <shell/shell.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <linker/linker-defs.h>
+
+void main(void)
+{
+	const struct device *uart_dev;
+	int ret;
+
+
+	/* Get uart_dev device */
+	uart_dev = device_get_binding("UART_1");
+	if (uart_dev == NULL) {
+		printk("Error: didn't find %x device\n", uart_dev);
+		return;
+	}
+	printk("Start the shell test!!!");
+}
+
+
+/******************************************************************************/
+/* Shell command functions */
+
+/* Memory test command help texts form Chromium EC */
+#define DUMP_CMD_HELP   "md [.b|.h|.s] addr [count]\n" \
+	"Dump memory values, optionally specifying the format"
+#define RW_CMD_HELP     "rw [.b|.h] addr [value]\n" \
+	"Read or write a word in memory optionally specifying the size"
+
+enum format {
+	FMT_WORD,
+	FMT_HALF,
+	FMT_BYTE,
+	FMT_STRING,
+};
+
+static void show_val(const struct shell *shell, uint32_t address,
+					uint32_t index, enum format fmt)
+{
+	uint32_t val;
+	uintptr_t ptr = address;
+
+	switch (fmt) {
+	case FMT_WORD:
+		if (0 == (index % 4))
+			shell_print(shell, "\n%08X:", address + index * 4);
+		val = *((uint32_t *)ptr + index);
+		shell_print(shell, " %08x", val);
+		break;
+	case FMT_HALF:
+		if (0 == (index % 8))
+			shell_print(shell, "\n%08X:", address + index * 2);
+		val = *((uint16_t *)ptr + index);
+		shell_print(shell, " %04x", val);
+		break;
+	case FMT_BYTE:
+		if (0 == (index % 16))
+			shell_print(shell, "\n%08X:", address + index);
+		val = *((uint8_t *)ptr + index);
+		shell_print(shell, " %02x", val);
+		break;
+	case FMT_STRING:
+		if (0 == (index % 32))
+			shell_print(shell, "\n%08X: ", address + index);
+		val = *((uint8_t *)ptr + index);
+		if (val >= ' ' && val <= '~')
+			shell_print(shell, "%c", val);
+		else
+			shell_print(shell, "\\x%02x", val);
+		break;
+	}
+}
+
+static int cmd_mem_dump(const struct shell *shell, size_t argc, char *argv[])
+{
+	uint32_t address, i, num = 1;
+	char *e;
+	enum format fmt = FMT_WORD;
+
+	if (argc > 1) {
+		if ((argv[1][0] == '.') && (strlen(argv[1]) == 2)) {
+			switch (argv[1][1]) {
+			case 'b':
```

Comment:
Why you will not create subcommands for this?

Reply:
Thanks for your comments. This PR is used for demonstrating how the shell works via npcx uart driver for our customers. This shell command is derived from the other RTOS. I really appreciate what you mentioned. Since I don't submit this PR to upstream. I will close it.